File: desktop/src/types/task.ts
export type TaskStatus =
  | 'created'
  | 'parsing'
  | 'analyzing'
  | 'waiting_confirm'
  | 'generating'
  | 'completed'
  | 'failed'

export interface TaskFile {
  id: string
  task_id: string
  original_filename: string
  object_key: string
  file_size: number
  content_type: string
  uploaded_at: string
}

export interface TaskListItem {
  id: string
  project_name: string
  remark: string | null
  source: string | null
  status: TaskStatus
  created_at: string
  updated_at: string
  file_count: number
}

export interface TaskDetail extends Omit<TaskListItem, 'file_count'> {
  parse_template_id: string | null
  files: TaskFile[]
}

export interface TaskListData {
  items: TaskListItem[]
  total: number
  page: number
  page_size: number
  pages: number
}

export interface TaskCreateInput {
  project_name: string
  remark?: string | null
  source?: string | null
  parse_template_id?: string | null
}

export interface DownloadUrlData {
  url: string
  expires_in: number
  filename: string
}

File: backend/app/schemas/task.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field, model_validator

from app.models.task import TaskStatus


class TaskCreate(BaseModel):
    project_name: str = Field(min_length=1, max_length=255)
    remark: str | None = None
    source: str | None = Field(default=None, max_length=255)
    parse_template_id: uuid.UUID | None = None


class TaskUpdate(BaseModel):
    project_name: str | None = Field(default=None, min_length=1, max_length=255)
    remark: str | None = None
    source: str | None = Field(default=None, max_length=255)
    parse_template_id: uuid.UUID | None = None

    @model_validator(mode="after")
    def reject_null_project_name(self) -> "TaskUpdate":
        if "project_name" in self.model_fields_set and self.project_name is None:
            raise ValueError("项目名称不能为 null")
        return self


class TaskStatusUpdate(BaseModel):
    status: TaskStatus


class TaskFileRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    task_id: uuid.UUID
    original_filename: str
    object_key: str
    file_size: int
    content_type: str
    uploaded_at: datetime


class TaskRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_name: str
    remark: str | None
    source: str | None
    parse_template_id: uuid.UUID | None
    status: TaskStatus
    created_at: datetime
    updated_at: datetime
    files: list[TaskFileRead] = Field(default_factory=list)


class TaskListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_name: str
    remark: str | None
    source: str | None
    status: TaskStatus
    created_at: datetime
    updated_at: datetime
    file_count: int = 0


class TaskListData(BaseModel):
    items: list[TaskListItem]
    total: int
    page: int
    page_size: int
    pages: int


class DownloadUrlData(BaseModel):
    url: str
    expires_in: int
    filename: str


class DeleteResult(BaseModel):
    id: uuid.UUID
    deleted: bool = True


class HealthData(BaseModel):
    status: str
    postgres: str
    redis: str
    minio: str
